From 13e132889d83d4b4b2eac08dcb87c2503838af84 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Fri, 9 Sep 2022 16:40:28 +0300
Subject: [PATCH 11/30] s32cc: Make PLL base addresses available for MMU

Make PLL base addresses available for MMU settings
through headers.

Issue: ALB-9287
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/nxp/s32/clk/early_clocks.c | 5 -----
 plat/nxp/s32/include/s32_clocks.h  | 9 +++++++++
 2 files changed, 9 insertions(+), 5 deletions(-)

diff --git a/drivers/nxp/s32/clk/early_clocks.c b/drivers/nxp/s32/clk/early_clocks.c
index 37f3386e9..f6376a384 100644
--- a/drivers/nxp/s32/clk/early_clocks.c
+++ b/drivers/nxp/s32/clk/early_clocks.c
@@ -16,11 +16,6 @@
 	.drv = &fake_clk_dev, \
 }
 
-#define ARM_PLL_BASE_ADDR		PLL_ADDR(S32_CORE_PLL)
-#define PERIPH_PLL_BASE_ADDR	PLL_ADDR(S32_PERIPH_PLL)
-#define ACCEL_PLL_BASE_ADDR		PLL_ADDR(S32_ACCEL_PLL)
-#define DRAM_PLL_BASE_ADDR		PLL_ADDR(S32_DDR_PLL)
-
 #define ARM_DFS_BASE_ADDR		S32_DFS_ADDR(S32_CORE_DFS)
 #define PERIPH_DFS_BASE_ADDR	S32_DFS_ADDR(S32_PERIPH_DFS)
 
diff --git a/plat/nxp/s32/include/s32_clocks.h b/plat/nxp/s32/include/s32_clocks.h
index e86a0608d..3a625b425 100644
--- a/plat/nxp/s32/include/s32_clocks.h
+++ b/plat/nxp/s32/include/s32_clocks.h
@@ -17,6 +17,15 @@
 #define CORE_PLL_BASE_ADDR	0x40038000ul
 #define PLL_ADDR(pll)		(CORE_PLL_BASE_ADDR + (pll) * 0x4000)
 
+#define ARM_PLL_BASE_ADDR	PLL_ADDR(S32_CORE_PLL)
+#define ARM_PLL_SIZE		(0x88)
+#define PERIPH_PLL_BASE_ADDR	PLL_ADDR(S32_PERIPH_PLL)
+#define PERIPH_PLL_SIZE		(0x100)
+#define ACCEL_PLL_BASE_ADDR	PLL_ADDR(S32_ACCEL_PLL)
+#define ACCEL_PLL_SIZE		(0x88)
+#define DRAM_PLL_BASE_ADDR	PLL_ADDR(S32_DDR_PLL)
+#define DRAM_PLL_SIZE		(0x84)
+
 #define PLL_MIN_FREQ		(1300000000ull)
 #define PLL_MAX_FREQ		(5000000000ull)
 
-- 
2.17.1

